src/java.xml/share/classes/com/sun/org/apache/xerces/internal/utils/XMLSecurityManager.java

```diff
@@ -561,8 +556,42 @@ private boolean getSystemProperty(Limit limit, String sysPropertyName) {
                 states[limit.ordinal()] = State.SYSTEMPROPERTY;
                 return true;
             }
+        } catch (NumberFormatException e) {
+            //invalid setting
+            throw new NumberFormatException("Invalid setting for system property: " + limit.systemProperty());
+        }
+        return false;
+    }
 
-            value = SecuritySupport.readJAXPProperty(sysPropertyName);
+    /**
+     * Reads the legacy system property.
+     * @param limit a limit object
+     * @return true if found, false otherwise
+     */
+    private boolean getOldSystemProperty(Limit limit) {
+        boolean found = false;
+        for (NameMap nameMap : NameMap.values()) {
+            String oldName = nameMap.getOldName(limit.systemProperty());
+            if (oldName != null) {
+                if (getSystemProperty(limit, oldName)) {
+                    found = true;
+                    break;
+                }
+            }
+        }
```

Comment:
It's a little hard to see what's going on here but this essentially boils down to a Map<String,String> mapping new names to old names. This or further refactoring should be considered.